### ISSUE TITLE
sieve_extension shall be evaluated only by timsieved, but lmtpd also tries to enfore it

### ISSUE THREAD

**Issue by dilyanpalauzov** [cyrusimap/cyrus-imapd#3227]
The documentation of imapd.conf:sieve_extension says
> Space-separated list of Sieve extensions allowed to be used in sieve scripts, enforced at submission by timsieved(8).  Any previously installed script will be unaffected by this option and will continue to execute regardless of the extensions used.  This option has no effect on options that are disabled at compile time (e.g., "regex").

lmdpd calls imap/lmtp_sieve.c:sieve_rebuild() which somehow verifies what is contained in sieve_extension.  On my system I get in the logs 
```
lmtpunix: sieve_rebuild: filename parse failed: script errors:
          line 3: Unsupported feature(s) in "require": "imapflags"
```
This indicates, that sieve_extension is not evaluated only by timsieved, when the script is uploaded, but can be enforced also by lmtpd.

**Comment by brong** [comment]
Yeah imapd.conf is wrong, we recompile sieve more often now.  I'll change imapd.conf in https://github.com/cyrusimap/cyrus-imapd/pull/5335
